src/Agent.Listener/CapabilitiesProvider.cs

```diff
@@ -0,0 +1,291 @@
+using Microsoft.VisualStudio.Services.Agent.Util;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.Services.Agent.Listener
+{
+    [ServiceLocator(Default = typeof(CapabilitiesProvider))]
+    public interface ICapabilitiesProvider : IAgentService
+    {
+        Task<Dictionary<string, string>> GetCapabilitiesAsync(string agentName, CancellationToken token);
+    }
+
+    public class Capability
+    {
+        public Capability(string name, string tool = null, string[] paths = null)
```

Comment:
There are 23 instances of Capability class and once instance of ToolCapability class. These classes are needed to have a typed structure that contains the configuration and improve readability. The other option is to use C# tuple which is less readable.